### ISSUE TITLE
Rolling file writer by date does not write in the original file

### ISSUE THREAD

**Issue by mlallaouret** [cihub/seelog#109]
Hello !

Given the following configuration:
```xml
<seelog type="sync" >
    <outputs formatid="main">
            <rollingfile type="date" datepattern="2006-02-02" filename="/var/log/logByDate.log" />
            <rollingfile type="size" maxsize="5000" filename="/var/log/logBySize.log" />
    </outputs>
    <formats>
        <format id="main" format="%Date(2006 Jan 02/3:04:05.000000000 PM MST) [%Level] %Msg%n"/>
    </formats>
</seelog>
```

Seelog will write current logs in logBySize.log as expected, but not in logByDate.log. Instead, it will write in the rolling file of the day: logByDate.log.2016-04-04. I suppose this is a bug, right ?

**Comment by goodsign** [comment]
Hey!

No, that works as intended: [rolling receiver](https://github.com/cihub/seelog/wiki/Receiver-reference#rolling-file-writer-or-rotation-file-writer)

Check the 'filename' param description.

**Comment by mlallaouret** [comment]
Ok, thank you for your answer.
So, having log files matching `filename.DATE_PATTERN` pattern, wouldn't it be better if the logs of the current period were always written in `filename` and just create the `filename.period` when seelog rolls the file ? It would be easier to access the log file of the current period (usually daily rotation) to make research, more recent logs always being in `filename.log`. Also, if I am not wrong, with the `size` type, seelog writes in `filename.log`, then in `filename.log.1`, `filename.log.2`, etc.... So the `size` type could also benefit from this feature.

**Comment by goodsign** [comment]
Yeah, probably it could be easier this way, I'll think about implementation possibility. Currently it doesn't check the file modification date and only checks the filename, thus we need to store the time in the filename to be able to know when to roll.

**Comment by jonsyu1** [comment]
Can we get the last modified time from the stat'ing the file and format that timestamp before diffing?

**Comment by goodsign** [comment]
Yeah, we can do that.

**Comment by jonsyu1** [comment]
Ah, this only works for `rollingIntervalAny`. I see that nothing sets custom rolling intervals at the moment except for tests. Are there plans to support this in the future?

If so, I think we'll need to write this information to disk. We can touch a `.seelog_last_roll` file on file creation during log roll and parse that to get `tprev` (previous roll time).

**Comment by goodsign** [comment]
The `rollingIntervalDaily` and related code is deprecated. It was used in earlier seelog versions and now it's just left there, I forgot to delete it. But it's basically unused, not mentioned in the [reference](https://github.com/cihub/seelog/wiki/Receiver-reference#rolling-file-writer-or-rotation-file-writer) and basically unusable, because the `rollingIntervalTypeFromString` func is not used anywhere. So it shouldn't affect anything and should be deleted. I'll go delete it now.

And yes, your initial solution works fine as far as I see, so maybe we should modify this [pull](https://github.com/cihub/seelog/pull/117) to implement the stat'n'diff solution, it should be easier than `.seelog_rolling_tracker` approach.

Basically, the [rolling writer](https://github.com/cihub/seelog/wiki/Receiver-reference#rolling-file-writer-or-rotation-file-writer) approach is pretty simple: we take the last modification time, format it using the `datepattern` parameter, then we take `time.LocalTime()`, format it using the `datepattern` parameter, then we compare and if results are different, we roll. Currently we store the already formatted modification time in the file. I think we could just use `stat` info and follow the same logic without changing the concept. This works fine.

**Comment by goodsign** [comment]
Ok, I removed the deprecated `rollingIntervalDaily`.

**Comment by jonsyu1** [comment]
Perfect, I'm on the same page here. Thanks for the quick response and cleanup. I'll go ahead and cleanup my end too.

**Comment by goodsign** [comment]
Fixed in latest revision. Thanks to @jonsyu1 for this [PR](https://github.com/cihub/seelog/pull/117)
